components/cors-mgt/org.wso2.carbon.identity.cors.mgt.core/src/main/java/org/wso2/carbon/identity/cors/mgt/core/internal/impl/CORSManagementServiceImpl.java

```diff
@@ -0,0 +1,333 @@
+                getConfigurationManager().deleteAttribute(CORS_ORIGIN_RESOURCE_TYPE_NAME, CORS_ORIGIN_RESOURCE_NAME,
+                        attribute.getKey());
+            }
+        } catch (ConfigurationManagementException | IOException e) {
+            throw handleServerException(ERROR_CODE_CORS_CRUD, e, tenantDomain, "deleting");
+        } finally {
+            FrameworkUtils.endTenantFlow();
+        }
+    }
+
+    /**
+     * Add the CORS Origin resource type to the ConfigurationManager if not already present.
+     *
+     * @throws ConfigurationManagementException
+     */
+    private void addCORSOriginResourceTypeIfNotExists() throws ConfigurationManagementException {
+
+        if (isCORSOriginResourceTypeNotExists()) {
+            ResourceTypeAdd resourceTypeAdd = createCORSOriginResourceTypeToAdd();
+            getConfigurationManager().addResourceType(resourceTypeAdd);
+        }
+    }
+
+    /**
+     * Returns true if the CORS Origin type is already in the ConfigurationManager.
+     *
+     * @return {@code true} if the CORS Origin resource type is already in the ConfigurationManager,
+     * {@code false} otherwise.
+     * @throws ConfigurationManagementException
+     */
+    private boolean isCORSOriginResourceTypeNotExists() throws ConfigurationManagementException {
+
+        try {
+            getConfigurationManager().getResourceType(CORS_ORIGIN_RESOURCE_TYPE_NAME);
+        } catch (ConfigurationManagementClientException e) {
+            if (ERROR_CODE_RESOURCE_TYPE_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
+                return true;
+            }
+            throw e;
+        }
+        return false;
+    }
+
+    /**
+     * Create a ResourceTypeAdd for CORS Origin type.
+     *
+     * @return ResourceTypeAdd A resource type with CORS_ORIGIN_RESOURCE_TYPE set as the name.
+     */
+    private ResourceTypeAdd createCORSOriginResourceTypeToAdd() {
+
+        ResourceTypeAdd resourceTypeAdd = new ResourceTypeAdd();
+        resourceTypeAdd.setName(CORS_ORIGIN_RESOURCE_TYPE_NAME);
+        resourceTypeAdd.setDescription(CORS_ORIGIN_RESOURCE_TYPE_DESCRIPTION);
+        return resourceTypeAdd;
+    }
+
+    /**
+     * Retrieve the ConfigurationManager instance from the CORSServiceHolder.
+     *
+     * @return ConfigurationManager The ConfigurationManager instance.
+     */
+    private ConfigurationManager getConfigurationManager() {
+
+        return CORSManagementServiceHolder.getInstance().getConfigurationManager();
+    }
+
+    /**
+     * Returns true if the tenant already has a particular CORS Origin.
+     *
+     * @param origin The Origin to be checked against the existing Origins.
+     * @return {@code true} if the tenant already have the particular CORS Origin, {@code false} otherwise.
+     * @throws ConfigurationManagementException
+     */
+    private boolean tenantHasCORSOrigin(CORSOrigin origin) throws ConfigurationManagementException, IOException {
```

Comment:
Can we rename this method? Something like 'isPreConfiguredCORSOrigin' or 'isDefinedCORSOrigin'

Reply:
Changed to `isDefinedCORSOriginResource`.